parameters/src/macros.rs

```diff
@@ -166,80 +166,86 @@ macro_rules! impl_load_bytes_logic_local {
 
 macro_rules! impl_load_bytes_logic_remote {
     ($remote_url: expr, $local_dir: expr, $filename: expr, $metadata: expr, $expected_checksum: expr, $expected_size: expr) => {
-        // Compose the correct file path for the parameter file.
-        let mut file_path = aleo_std::aleo_dir();
-        file_path.push($local_dir);
-        file_path.push($filename);
-
-        let buffer = if file_path.exists() {
-            // Attempts to load the parameter file locally with an absolute path.
-            std::fs::read(&file_path)?
-        } else {
-            // Downloads the missing parameters and stores it in the local directory for use.
-             #[cfg(not(feature = "no_std_out"))]
-            {
-                use colored::*;
-                let path = format!("(in {:?})", file_path);
-                eprintln!(
-                    "\n⚠️  \"{}\" does not exist. Downloading and storing it {}.\n",
-                    $filename, path.dimmed()
-                );
-            }
-
-            // Construct the URL.
-            let url = format!("{}/{}", $remote_url, $filename);
-
-            // Load remote file
-            cfg_if::cfg_if! {
-                if #[cfg(not(feature = "wasm"))] {
-                    let mut buffer = vec![];
-                    Self::remote_fetch(&mut buffer, &url)?;
-
-                    // Ensure the checksum matches.
-                    let candidate_checksum = checksum!(&buffer);
-                    if $expected_checksum != candidate_checksum {
-                        return checksum_error!($expected_checksum, candidate_checksum)
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "filesystem")] {
```

Comment:
```suggestion
            if #[cfg(feature = "filesystem")] {
```

This creates the need to keep track of always including the filesystem feature in any SnarkVM crates that depend directly on parameters. But perhaps more damaging to the ecosystem is that sometimes external rust crates DO in fact depend on the parameters crate [example](https://github.com/ProvableHQ/sdk/blob/mainnet/wasm/Cargo.toml#L45-L47) in ways that don't simply assume the default features, and this could take down services upon an upgrade.

I agree with @vicsn that being able to specify whether you have a filesystem is a good idea, but I think what would be better since it's much larger subset of people who will be using filesystems than not, is to create a `no-filesystem` feature flag and use instead as it would have fewer downstream side effects.